explainability-service/src/main/java/org/kie/trustyai/service/endpoints/metrics/DisparateImpactRatioEndpoint.java

```diff
@@ -114,7 +121,16 @@ public Response getDefinition() {
     @Produces(MediaType.APPLICATION_JSON)
     @Path("/definition")
     public Response getSpecificDefinition(DefinitionRequest request) {
-        return Response.ok(calculator.getDIRDefinition(request.getMetricValue(), request)).build();
+        final ReconciledMetricRequest reconciledMetricRequest;
+        try {
+            reconciledMetricRequest = ReconciledMetricRequest.reconcile(request, dataSource);
```

Comment:
Why the `reconcile` method here is invoked in a `try/catch` while in the `dir` method this is not the case?
Are you sure the `DataframeCreateException` cannot happen in that method too?

Reply:
In this function, I'm calling:

`ReconciledMetricRequest.reconcile(rawRequest, dataSource)`

In `dir`, I'm calling:

 `ReconciledMetricRequest.reconcile(rawRequest, metadata)`
 
 The `DataframeCreateException` can be thrown on the creation of the metadata, which happens only in the first invocation.